src/traces/scattergeo/plot.js

```diff
@@ -125,83 +125,84 @@ plotScatterGeo.plot = function(geo, scattergeoData) {
 
-            d3.select(this)
-                .append('path')
-                .datum(makeLineGeoJSON(trace))
-                .attr('class', 'js-line');
-        });
+    gScatterGeoTraces.each(function(trace) {
+        var s = d3.select(this);
 
-    gScatterGeoTraces.append('g')
-        .attr('class', 'points')
-        .each(function(trace) {
-            var s = d3.select(this),
-                showMarkers = subTypes.hasMarkers(trace),
-                showText = subTypes.hasText(trace);
-
-            if((!showMarkers && !showText)) return;
-
-            var cdi = plotScatterGeo.calcGeoJSON(trace, geo.topojson),
-                cleanHoverLabelsFunc = makeCleanHoverLabelsFunc(geo, trace),
-                eventDataFunc = makeEventDataFunc(trace);
-
-            var hoverinfo = trace.hoverinfo,
-                hasNameLabel = (
-                    hoverinfo === 'all' ||
-                    hoverinfo.indexOf('name') !== -1
-                );
-
-            function handleMouseOver(pt, ptIndex) {
-                if(!geo.showHover) return;
-
-                var xy = geo.projection([pt.lon, pt.lat]);
-                cleanHoverLabelsFunc(pt);
-
-                Fx.loneHover({
-                    x: xy[0],
-                    y: xy[1],
-                    name: hasNameLabel ? trace.name : undefined,
-                    text: pt.textLabel,
-                    color: pt.mc || (trace.marker || {}).color
-                }, {
-                    container: geo.hoverContainer.node()
-                });
-
-                geo.graphDiv.emit('plotly_hover', eventDataFunc(pt, ptIndex));
-            }
-
-            function handleClick(pt, ptIndex) {
-                geo.graphDiv.emit('plotly_click', eventDataFunc(pt, ptIndex));
-            }
-
-            if(showMarkers) {
-                s.selectAll('path.point')
-                    .data(cdi)
-                    .enter().append('path')
-                        .attr('class', 'point')
-                        .on('mouseover', handleMouseOver)
-                        .on('click', handleClick)
-                        .on('mouseout', function() {
-                            Fx.loneUnhover(geo.hoverContainer);
-                        })
-                        .on('mousedown', function() {
-                            // to simulate the 'zoomon' event
-                            Fx.loneUnhover(geo.hoverContainer);
-                        })
-                        .on('mouseup', handleMouseOver);  // ~ 'zoomend'
-            }
-
-            if(showText) {
-                s.selectAll('g')
-                    .data(cdi)
-                    .enter().append('g')
-                        .append('text');
-            }
-        });
+        if(!subTypes.hasLines(trace)) return;
+
+        s.selectAll('path.js-line')
+            .data([makeLineGeoJSON(trace)])
+          .enter().append('path')
+            .classed('js-line', true);
+
+        // TODO add hover - how?
```

Comment:
Line 188 handles hover for markers, not lines unfortunately.

That will come when someone has the guts to refactor `graph_interact.js` so that it can be used with arbitrary pixel-to-coordinate transforms.